components/FormInput/SharedFormInput.js

```diff
@@ -60,21 +60,46 @@ export default class SharedFormInput extends Component {
       error,
       value,
       label,
+      labelOutside,
       ...remainingProps,
     } = this.props;
+
     const { focus, id } = this.state;
 
-    const active = value && !!value.length;
-    const isInsideVariant = !!placeholder;
+    const hasValue = !!value && !!value.length;
+    const labelIsOutside = labelOutside || !placeholder;
+    const labelIsOutsideWithPlaceholder = labelOutside && !!placeholder;
+
+    const labelPositionClasses = [];
+
+    if (labelIsOutside) {
+      labelPositionClasses.push(css.labelOutside);
+    } else {
+      labelPositionClasses.push(css.labelInside);
+    }
+
+    if (labelIsOutside && labelIsOutsideWithPlaceholder) {
+      labelPositionClasses.push(css.labelOutsideWithPlaceholder);
+    }
 
     const outerCSS = [
       css.container,
-      required ? css.required : null,
-      isInsideVariant ? css.labelInside : css.labelOutside,
-      error ? css.error : null,
-      active ? css.active : null,
-      focus ? css.focus : null,
-      borderless ? css.borderless : null,
+      required ?
+        css.required :
+        null,
```

Comment:
Hmm, I went with keeping `?` on the same line as it's the indicator of what to expect from the next line, but I'm not fussed. Will change